deploy-service/common/src/main/java/com/pinterest/deployservice/rodimus/RodimusManagerImpl.java

```diff
@@ -73,40 +72,24 @@ public boolean shouldSkipClass(Class<?> c) {
         }
     }
 
-    private boolean refreshCachedKey() throws Exception {
-        String prevKnoxKey = this.cachedKey;
-
-        if (this.fsKnox != null) {
-            this.cachedKey = new String(this.fsKnox.getPrimaryKey());
-        }
-
-        if ( prevKnoxKey == null )
-        {
-            return this.cachedKey != null;
-        }else{
-            return ! prevKnoxKey.equals( this.cachedKey );
-        }
-    }
-
     private void setAuthorization() throws Exception {
-        if (this.cachedKey == null) {
-            this.refreshCachedKey();
-        }
-        this.headers.put("Authorization", String.format("token %s", this.cachedKey));
+        String currKnoxKey = new String(this.fsKnox.getPrimaryKey());
+        String rodimusKey = currKnoxKey.replaceAll("[\n\r]*$", "");
```

Comment:
Can you please use a non-regex approach? Also, this method is deprecated and moved, so please do not use it.  You can use any non-regex method to replace all of a certain character with ''.

https://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/StringUtils.html#replaceAll-java.lang.String-java.lang.String-java.lang.String-

Reply:
From discussion: it is ok to use the regex if you simplify it to only matching the 3 desired characters.

Reply:
done